zookeeper-recipes/zookeeper-recipes-queue/src/test/java/org/apache/zookeeper/recipes/queue/DistributedQueueTest.java

```diff
@@ -27,243 +28,260 @@
+      throws Exception {
+    String testString = "Hello World";
+    final int numClients = 2;
+    ZooKeeper[] clients = new ZooKeeper[numClients];
+    DistributedQueue[] queueHandles = new DistributedQueue[numClients];
+    for (int i = 0; i < clients.length; i++) {
+      clients[i] = createClient();
+      queueHandles[i] = new DistributedQueue(clients[i], dir, null);
+    }
 
-        Assert.assertNotNull(data);
-        Assert.assertEquals(new String(data), testString + (m - 1));
+    for (int i = 0; i < n; i++) {
+      String offerString = testString + i;
+      queueHandles[0].offer(offerString.getBytes(UTF_8));
     }
 
-    @Test
-    public void testRemove2() throws Exception {
-        createNremoveMtest("/testRemove2", 10, 2);
+    byte[] data = null;
+    for (int i = 0; i < m; i++) {
+      data = queueHandles[1].remove();
     }
-    @Test
-    public void testRemove3() throws Exception {
-        createNremoveMtest("/testRemove3", 1000, 1000);
+    Assert.assertEquals(new String(queueHandles[1].element(), UTF_8),
+        testString + m);
+  }
+
+  @Test
+  public void testElement1() throws Exception {
+    createNremoveMelementTest("/testElement1", 1, 0);
+  }
+
+  @Test
+  public void testElement2() throws Exception {
+    createNremoveMelementTest("/testElement2", 10, 2);
+  }
+
+  @Test
+  public void testElement3() throws Exception {
+    createNremoveMelementTest("/testElement3", 1000, 500);
+  }
+
+  @Test
+  public void testElement4() throws Exception {
+    createNremoveMelementTest("/testElement4", 1000, 1000 - 1);
+  }
+
+  @Test
+  public void testTakeWait1() throws Exception {
+    String dir = "/testTakeWait1";
+    final String testString = "Hello World";
+    final int numClients = 1;
+    final ZooKeeper[] clients = new ZooKeeper[numClients];
+    final DistributedQueue[] queueHandles = new DistributedQueue[numClients];
+    for (int i = 0; i < clients.length; i++) {
+      clients[i] = createClient();
+      queueHandles[i] = new DistributedQueue(clients[i], dir, null);
     }
 
-    public void createNremoveMelementTest(String dir, int n, int m) throws Exception {
-        String testString = "Hello World";
-        final int numClients = 2;
-        ZooKeeper[] clients = new ZooKeeper[numClients];
-        DistributedQueue[] queueHandles = new DistributedQueue[numClients];
-        for (int i = 0; i < clients.length; i++) {
-            clients[i] = createClient();
-            queueHandles[i] = new DistributedQueue(clients[i], dir, null);
-        }
+    final byte[] takeResult[] = new byte[1][];
+    Thread takeThread = new Thread() {
+      public void run() {
+        try {
+          takeResult[0] = queueHandles[0].take();
+        } catch (KeeperException e) {
 
-        for (int i = 0; i < n; i++) {
-            String offerString = testString + i;
-            queueHandles[0].offer(offerString.getBytes());
-        }
+        } catch (InterruptedException e) {
```

Comment:
Why separate out these exceptions instead of keeping the original multi-catch? This seems like a regression.